packages/block-library/src/navigation-link/test/edit.js

```diff
@@ -0,0 +1,506 @@
+			} );
+		} );
+
+		it( 'can update a custom tag link', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 4,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'PortfolioTag',
+				type: 'portfolio_tag',
+				url: 'http://wordpress.local/portfolio_tag/portfoliotag/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				id: 4,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'PortfolioTag',
+				type: 'portfolio_tag',
+				url: 'http://wordpress.local/portfolio_tag/portfoliotag/',
+			} );
+		} );
+
+		it( 'can update a custom category link', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 2,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'Portfolioz Category',
+				type: 'portfolio_category',
+				url:
+					'http://wordpress.local/portfolio_category/portfolioz-category/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				id: 2,
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'Portfolioz Category',
+				type: 'portfolio_category',
+				url:
+					'http://wordpress.local/portfolio_category/portfolioz-category/',
+			} );
+		} );
+
+		it( 'can update a post format', () => {
+			const setAttributes = jest.fn();
+			const linkSuggestion = {
+				id: 'video',
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				title: 'Video',
+				type: 'post-format',
+				url: 'http://wordpress.local/type/video/',
+			};
+			updateNavigationLinkBlockAttributes(
+				linkSuggestion,
+				setAttributes
+			);
+			expect( setAttributes ).toHaveBeenCalledWith( {
+				kind: 'taxonomy',
+				opensInNewTab: false,
+				label: 'Video',
+				type: 'post_format',
+				url: 'http://wordpress.local/type/video/',
+			} );
+		} );
+
+		describe( 'various link protocols save as custom links', () => {
+			it( 'when typing a url, but not selecting a search suggestion', () => {
+				const setAttributes = jest.fn();
+				const linkSuggestion = {
+					opensInNewTab: false,
+					url: 'www.wordpress.org',
+				};
+				updateNavigationLinkBlockAttributes(
+					linkSuggestion,
+					setAttributes
+				);
+				expect( setAttributes ).toHaveBeenCalledWith( {
+					opensInNewTab: false,
+					url: 'www.wordpress.org',
+					label: 'www.wordpress.org',
+					kind: 'custom',
+				} );
+			} );
+
+			it( 'url', () => {
+				const setAttributes = jest.fn();
+				const linkSuggestion = {
+					id: 'www.wordpress.org',
+					opensInNewTab: false,
+					title: 'www.wordpress.org',
+					type: 'URL',
+					url: 'http://www.wordpress.org',
+				};
+				updateNavigationLinkBlockAttributes(
+					linkSuggestion,
+					setAttributes
+				);
+				expect( setAttributes ).toHaveBeenCalledWith( {
+					opensInNewTab: false,
+					label: 'www.wordpress.org',
+					kind: 'custom',
```

Comment:
I've just thought, by stripping out the data about whether this is a `tel`, `URL`, `#anchor`...etc are we going to cause problems if we ever want to render the "type" in the `LinkPreview` so the user knows this is a `telephone` link...etc?

We currently visually distinguish `tel`, `hash`...etc from each other in the search suggestions and we are going to lose this data as soon as the suggestion is selected.

Is there anywhere else we could store it?

I know I said it was ok to transform them to `custom` but thinking about it we might need to discuss options for being more nuanced. Apologies!

Reply:
I wonder if the `type` attribute could store this info? Should be safe as the nav_menu_item ignores `type` if `kind` is `custom`.

Reply:
It doesn't hurt to persist extra data here. I tried this in 5d19a86 for everything but "URL" since that felt redundant. 

One downside is that it ties together LinkControl typing that doesn't necessarily match our menu data models, but with the extra information here, we can choose to transform it later.